jerry-core/parser/js/js-parser.c

```diff
@@ -1350,31 +1355,42 @@ parser_post_processing (parser_context_t *context_p) /**< context */
   }
 
 #if ENABLED (JERRY_ESNEXT)
+  uint16_t function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_NORMAL);
+
   if (context_p->status_flags & (PARSER_IS_PROPERTY_GETTER | PARSER_IS_PROPERTY_SETTER))
   {
-    compiled_code_p->status_flags |= CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_ACCESSOR);
+    function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_ACCESSOR);
   }
   else if (context_p->status_flags & PARSER_IS_ARROW_FUNCTION)
   {
-    compiled_code_p->status_flags |= CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_ARROW);
+    function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_ARROW);
   }
   else if (context_p->status_flags & PARSER_IS_GENERATOR_FUNCTION)
   {
-    compiled_code_p->status_flags |= CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_GENERATOR);
+    if (context_p->status_flags & PARSER_IS_ASYNC_FUNCTION)
+    {
+      function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_ASYNC_GENERATOR);
+    }
+    else
+    {
+      function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_GENERATOR);
+    }
   }
   else if (context_p->status_flags & PARSER_IS_ASYNC_FUNCTION)
   {
-    compiled_code_p->status_flags |= CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_ASYNC);
+    function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_ASYNC);
   }
   else if (context_p->status_flags & PARSER_CLASS_CONSTRUCTOR)
   {
-    compiled_code_p->status_flags |= CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_CONSTRUCTOR);
+    function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_CONSTRUCTOR);
   }
   else
   {
-    compiled_code_p->status_flags |= CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_NORMAL);
+    function_type = CBC_FUNCTION_TO_TYPE_BITS (CBC_FUNCTION_NORMAL);
   }
 
+  compiled_code_p->status_flags |= function_type;
```

Comment:
How about using `CBC_FUNCTION_TO_TYPE_BITS` only here?

Reply:
I have been thinking about this, but that would mean an actual runtime computation (extra CPU instructions for shifting). The current method puts a final constant directly into a variable. The compiler might be able optimize the unnecessary shift, but I am not sure about it.

Reply:
I think this is not a 'hot' code path and the suggestion results slightly smaller binary size. So I'd still support the suggested changes.

Reply:
I've realizes that these are constants. Just ignore my comments.